Foo 062 (#260)
resource-watch/data-pre-processing: foo_062_rw0_fishery_production/README.md

> # FISHERY PRODUCTION

Reviewer: ```suggestion
# Fishery Production
```

The source provided this data in three datasets, global (total) fishery production as well as disaggregated data for aquaculture and capture production. The data on global production and capture each provide a CSV providing the quanity of production (tonnes). The data on aquaculture provides two CSVs, on for the quanity of production (tonnes) and the value of production (1000 USD). The data was provided within three zip folders. 

Below we describe the main steps taken to process the data so that it is formatted correctly to be uploaded to Carto.

1. Unzip each  folder and read in the dataset and the country code list as a pandas dataframe.
2. Rename the 'UN_Code' column in the country code list to 'COUNTRY.UN_CODE' so it matches the column header in the dataset.
3. Merge the country code list to the dataset so each row in the dataset is matched with an ISO code and its full name.
4. Add a column to reflect the type of production measured by the value column for the dataset (ex GlobalProduction, Aquaculture, or Capture) and the variable measured (quantity or value).

> 5. Convert the 'VALUE' column to a float.

Reviewer: ```suggestion
5. Convert the data type of the 'VALUE' column to float.
```

6. Concatenate the three dataframes for Global Production, Aquaculture, and Capture. 

> 7. Rename the 'PERIOD' column to 'year.'

Reviewer: ```suggestion
7. Rename the 'PERIOD' column to 'year'.
```

> 8. Pivot the dataframe from long to short to with country and year as the index, the 'type' of production as the columns, and summed values for each type of production in a given year.

Reviewer: ```suggestion
8. Pivot the dataframe from long to wide form to sum the values for each type of production in a given year for each country.
```
